internal/envoy/config/hcm.go

```diff
@@ -239,13 +239,11 @@ func ConfigSource(cluster string) *envoy_core_v3.ConfigSource {
 		ResourceApiVersion: envoy_core_v3.ApiVersion_V3,
 		ConfigSourceSpecifier: &envoy_core_v3.ConfigSource_ApiConfigSource{
 			ApiConfigSource: &envoy_core_v3.ApiConfigSource{
-				TransportApiVersion:       envoy_core_v3.ApiVersion_V3,
-				ApiType:                   envoy_core_v3.ApiConfigSource_GRPC,
-				RequestTimeout:            durationpb.New(defaultRequestTimeout),
-				SetNodeOnFirstMessageOnly: true,
-				GrpcServices: []*envoy_core_v3.GrpcService{
-					makeGrpcService(cluster, "", defaultResponseTimeout),
-				},
+				TransportApiVersion: envoy_core_v3.ApiVersion_V3,
+				ApiType:             envoy_core_v3.ApiConfigSource_GRPC,
+				RequestTimeout:      durationpb.New(defaultRequestTimeout),
+				GrpcServices:        []*envoy_core_v3.GrpcService{makeGrpcService(cluster, "", defaultResponseTimeout)},
+				//SetNodeOnFirstMessageOnly: true, // Could someone please justify why this is set to `true`?
```

Comment:
I'm not entirely sure myself, but it seems to make a difference in secrets being discovered or not, what I mean is, if I leave this field in place, the secrets stays in `dynamic_warming_secrets` object instead of the `dynamic_active_secrets` object.

Reply:
@jasmingacic and @povilasv, any idea?